### PR TITLE
udevadm: add --wait-for-initialization option and use it in systemd-networkd-tests.py

### PR DESCRIPTION
**Pull request by yuwata**
And add more tests about networkctl.

(This intends to detect the CI failure in #12700 easier.)

### CODE REVIEW

**Comment by ssahani** [comment]
@yuwata we can replace iproute with networkctl delete.

**Comment by yuwata** [comment]
Yeah, I forgot that. Will update.

**Comment by ssahani** [comment]
Great thanks

**Comment by yuwata** [comment]
@mrc0mmand CentOS CI (Arch in KVM) does not have `99-default.link`?
```
test_udev (__main__.NetworkctlTests) ... 
● 29: test1                                                  
             Link File: n/a                                  
          Network File: /run/systemd/network/11-dummy.network
                  Type: ether                                
                 State: degraded (configured)                
                Driver: dummy                                
            HW Address: 12:19:6a:d0:ba:df(null)              
                   MTU: 1500 (Minimum: 0, Maximum: 0)        
  Queue Length (Tx/Rx): 1/1                                  
               Address: fe80::1019:6aff:fed0:badf
FAIL
```

**Comment by mrc0mmand** [comment]
@yuwata it looks like it's there:

```
drwxr-xr-x  2 root root 4096 May 31 01:33 .
drwxr-xr-x 18 root root 4096 May 31 01:33 ..
-rw-r--r--  1 root root  645 May 31 01:26 80-container-host0.network
-rw-r--r--  1 root root  718 May 31 01:26 80-container-ve.network
-rw-r--r--  1 root root  704 May 31 01:26 80-container-vz.network
-rw-r--r--  1 root root  441 May 31 01:26 99-default.link
#  SPDX-License-Identifier: LGPL-2.1+
#
#  This file is part of systemd.
#
#  systemd is free software; you can redistribute it and/or modify it
#  under the terms of the GNU Lesser General Public License as published by
#  the Free Software Foundation; either version 2.1 of the License, or
#  (at your option) any later version.

[Match]
OriginalName=*

[Link]
NamePolicy=keep kernel database onboard slot path
MACAddressPolicy=persistent
```

**Comment by yuwata** [comment]
Thanks. So, it seems a udev's bug...

**Comment by keszybz** [comment]
We should probably figure out the CI failure before merging...

**Comment by yuwata** [comment]
Yeah...

**Comment by yuwata** [comment]
@keszybz Do you have any idea why it fails? The driver field is set, so `net_setup_link` builtin is correctly called, but it seems not find the .link file.

**Comment by yuwata** [comment]
`test_udev()` was dropped. It should be now safe now to be merged.

**Comment by yuwata** [comment]
@mrc0mmand Which *SAN requires `SystemCallFilter=`? Now at least one `--*san-options` is set, then the system call filter will be reset.

**Comment by mrc0mmand** [comment]
@yuwata LSan had issues with the current `SystemCallFilter`:

```
May 30 14:57:05 arch.localdomain systemd[1]: Stopping Network Service...
May 30 14:57:05 arch.localdomain systemd-networkd[11940]: ==11940==LeakSanitizer has encountered a fatal error.
May 30 14:57:05 arch.localdomain systemd-networkd[11940]: ==11940==HINT: For debugging, try setting environment variable LSAN_OPTIONS=verbosity=1:log_threads=1
May 30 14:57:05 arch.localdomain systemd-networkd[11940]: ==11940==HINT: LeakSanitizer does not work under ptrace (strace, gdb, etc)
```

so this should make it go away without tweaking the service file in CentOS CI. Thanks!

**Comment by yuwata** [comment]
Yey! Now all CIs are green!

File: man/udevadm.xml [hunk 1]
```diff
@@ -175,6 +175,13 @@
             <para>Cleanup the udev database.</para>
           </listitem>
         </varlistentry>
+        <varlistentry>
+          <term><option>-w</option></term>
+          <term><option>--wait-for-initialization<optional>=SECONDS</optional></option></term>
```
Comment (keszybz) [inline man/udevadm.xml:180]: We also have `udevadm trigger -w/--settle`. Those two options have similar meaning, so I guess that using `-w` for both of them is reasonable.

File: man/udevadm.xml [hunk 2]
```diff
@@ -175,6 +175,13 @@
...
+          <listitem>
+            <para>Wait for device to be initialized.</para>
```
Comment (keszybz) [inline man/udevadm.xml]: This says nothing about the default. I think this should say `"If argument <replaceable>seconds</replaceable> is not specified, the default is to wait forever."`.

File: src/udev/udevadm-info.c [hunk 3]
```diff
@@ -40,6 +41,8 @@ typedef enum QueryType {
 static bool arg_root = false;
 static bool arg_export = false;
 static const char *arg_export_prefix = NULL;
+static bool arg_wait_for_initialization = false;
```
Comment (keszybz) [inline src/udev/udevadm-info.c]: I don't think a separate `bool` is needed. Just initialize `arg_wait_for_initialization` to 0.

File: src/udev/udevadm-info.c [hunk 4]
```diff
@@ -414,6 +420,14 @@ int info_main(int argc, char *argv[], void *userdata) {
                         arg_export = true;
                         arg_export_prefix = optarg;
                         break;
+                case 'w':
+                        arg_wait_for_initialization = true;
+                        if (optarg) {
+                                r = parse_sec(optarg, &arg_wait_for_initialization_timeout);
+                                if (r < 0)
+                                        return log_error_errno(r, "Failed to parse timeout value: %m");
+                        }
```
Comment (keszybz) [inline src/udev/udevadm-info.c]: If no timeout is specified, I think the default value should be infinity, not some magic value.

**Comment by yuwata** [comment]
@keszybz Thank you for your review. I've force-pushed an updated version which I hope addresses your comments. PTAL.

**Pull request merged**